File: fake_names.js
var names = [
  'Benedikt Vlach',
  'Soběslav Fišer',
  'Bartoloměj Švec',
  'Radim Dostál',
  'Melichar Žák',
  'Patrik Adamec',
  'Vlastislav Havel',
  'Josef Rmoutil',
  'Vítězslav Dostál',
  'Adolf Šulc',
  'Vlastislav Špaček',
  'Zbyšek Slavík',
  'Vojtěch Dvořáček',
  'Teodor Zapletal',
  'Leoš Vlček',
  'Lukáš Strnad',
  'Denis Charvát',
  'Petr Povýšil',
  'Rostislav Vykukal',
  'Kvido Hanák'
];

function getAFakeDate(start_year, max_year) {
  var year = Math.abs(Math.floor(Math.random() * (max_year - start_year)) + start_year);
  var month = Math.floor(Math.random() * 11) + 1;
  var day = Math.floor(Math.random() * 27) + 1;
  var dt = new Date(year, month, day);
  return dt;
}

function getAFake() {
    var n = Math.floor(Math.random() * names.length);
    var p = {'fname':names[n]};
    var birth = getAFakeDate(1850, 2000);
    var death = getAFakeDate(birth.getFullYear(), 2019);

    p['birth'] = birth;
    p['death'] = death;
    return p;
}

/* This is just here to provide some dummy data */
function populateNames2() {
  baseCount = 100;
  rowCount = 10;   // number of rows from the gate to the pavillion
  groupCount = 10; // plot groups per row
  plotCount = 2;   // plots in each plot group

  thePlots = document.getElementById("the-plots");
  for (r = 1; r <= rowCount; r++) {
    row = document.createElement('div');
    row.id = "R" + r;
    row.setAttribute("class", 'plot-row');
    thePlots.appendChild(row);
    for (pg = 1; pg <= groupCount; pg++) {
      group = document.createElement('div');
      group.id = row.id + "C" + pg;
      group.setAttribute("class", 'plot-group');
      row.appendChild(group);
      for (p = 1; p <= plotCount; p++) {
        plot = document.createElement('div');

        if (p == 2) {
          plot.setAttribute("class", 'plot-left');
          plot.id = group.id + "P" + "L";

        } else {
          plot.setAttribute("class", 'plot-right');
          plot.id = group.id + "P" + "R";
        }

        label = document.createTextNode("");

        plot.appendChild(label);
        group.appendChild(plot);
        // console.log(plot.id);
        // get the appropriate contents (should be a call to a function)
        plotinfo = getContents(plot.id);
        if (!plotinfo) {
           plotinfo = getAFake();
           plotinfo['plotId'] = plot.id;
           plots[plot.id] = plotinfo;
        }

				if (!plotinfo) {
          console.log("Problem with plot");
          break;
        }

        // This line determines what data is put on the plot. Right now it is configured for the second word in the name.
        if (plotinfo['lname']) {
          plot.innerHTML = plotinfo['lname'];
        } else {
          plot.innerHTML = plotinfo['plotId'];
          plot.setAttribute('class','plot-empty');
        }
        plot.style.zindex = 50;
        plot.onmouseover = function(){ showDetails(this); };
      }
    }
  }
  // console.log("Appended " + thePlots.childElementCount  + " rows to " + row.parentNode);
}

File: fake_names.py
#! /bin/env python

import sys
import math
import random
import datetime
import json
import csv

first_names = [
  'Benedikt',
  'Soběslav',
  'Bartoloměj',
  'Radim',
  'Melichar',
  'Patrik',
  'Vlastislav',
  'Josef',
  'Vítězslav',
  'Adolf',
  'Vlastislav',
  'Zbyšek',
  'Vojtěch',
  'Teodor',
  'Leoš',
  'Lukáš',
  'Denis',
  'Petr',
  'Rostislav',
  'Kvido'
]
last_names = [
  'Vlach',
  'Fišer',
  'Švec',
  'Dostál',
  'Žák',
  'Adamec',
  'Havel',
  'Rmoutil',
  'Dostál',
  'Šulc',
  'Špaček',
  'Slavík',
  'Dvořáček',
  'Zapletal',
  'Vlček',
  'Strnad',
  'Charvát',
  'Povýšil',
  'Vykukal',
  'Hanák'
]

def getAFakeDate(start_year, max_year):
  year = abs(math.floor(random.random() * (max_year - start_year)) + start_year)
  month = math.floor(random.random() * 11) + 1
  day = math.floor(random.random() * 27) + 1
  dt = datetime.date(year, month, day)
  return dt

def getAFake():
    n = math.floor(random.random() * len(first_names))
    m = math.floor(random.random() * len(last_names))
    p ={'name':first_names[n] + " " + last_names[m]}
    birth = getAFakeDate(1850, 2000)
    death = getAFakeDate(birth.year, 2019)

    p['birth'] = birth.strftime("%x")
    p['death'] = death.strftime("%x")
    return p

if __name__ == '__main__':
    profiles = []
    for r in range(1, 6):
      for c in range(1, 11):
        for p in range(1, 5):
          profiles.append(getAFake())
          profiles[-1]['location'] = 'R'+str(r)+'C'+str(c)+'P'+str(p)
          profiles[-1]['description'] = 'blurb'
    fieldnames = list(profiles[0].keys())
    w = csv.DictWriter(sys.stdout, fieldnames= fieldnames, quoting=csv.QUOTE_NONNUMERIC)
    w.writeheader()
    for row in profiles:
       w.writerow(row) 
    #print(json.dumps(profiles))
